### PR TITLE
NP-45840 Aggregate content files to records with same cristin id

### PR DESCRIPTION
**Pull request by joachimjorgensen**
When there are (unit content) rows with duplicate cristin id, the files are aggregated to one record. 
A caveat is that PublishingAuthority can be different from file to file, but the value is set directly on the Record-object. Thus, PublishingAuthority will be set to the value of the first unis content file processed.

### CODE REVIEW

File: src/main/java/no/sikt/nva/UnisContentScrapingResult.java [hunk 1]
```diff
@@ -34,23 +38,36 @@ public static UnisContentScrapingResult fromWorkbook(Workbook workbook)
...
 
-    public List<Record> getResults() {
-        return this.results;
+    private static void processUnisContent(Map<String, Record> processedRecords, UnisContent unisContent)
+        throws URISyntaxException {
+
+        var newRecord = unisContent.toRecord();
+        if (processedRecords.containsKey(newRecord.getCristinId())) {
+            var existingRecord = processedRecords.get(newRecord.getCristinId());
+            var newContentFile = newRecord.getContentBundle().getContentFiles().get(0);
```
Comment (torbjokv) [inline src/main/java/no/sikt/nva/UnisContentScrapingResult.java:66]: I don't know anything about the unis stuff here, but is it always one and only one file here?

File: src/main/java/no/sikt/nva/UnisContent.java [hunk 2]
```diff
@@ -127,7 +128,7 @@ var record = new Record();
             fileIdentifier,
             License.fromBrageLicense(getLicense()),
             getEmbargo());
-        record.setContentBundle(new ResourceContent(List.of(contentFile)));
+        record.setContentBundle(new ResourceContent(new ArrayList<>(List.of(contentFile))));
```
Comment (torbjokv) [inline src/main/java/no/sikt/nva/UnisContent.java:131]: assume new ArrayList<> here is to make the list mutable, maybe move it inside resourcecontent constructor?

File: src/test/java/no/sikt/nva/UnisContentScrapingResultTest.java [hunk 3]
```diff
@@ -98,4 +99,46 @@ void shouldThrowExceptionIfUnknownColumnHeaderExists() {
                      () -> UnisContentScrapingResult.fromWorkbook(createWorkbook(invalidExcelData)),
                      ERROR_MESSAGE_INVALID_HEADERS);
     }
+
+    @Test
+    void shouldAppendContentFilesToSameRecordWhenCristinIdIsIdentical()
```
Comment (torbjokv) [inline src/test/java/no/sikt/nva/UnisContentScrapingResultTest.java:104]: split this test maybe?

**Review by torbjokv — COMMENTED** [review]
someone that know the domain here should also look at this pr, but otherwise I think it looks ok with only some suggestions :-)

File: src/main/java/no/sikt/nva/UnisContentScrapingResult.java (hunk 1 above)
Comment (anetteOlli) [reply src/main/java/no/sikt/nva/UnisContentScrapingResult.java:66]: Torbrjørn: NVA support multiple files per cristin-id. What Unis does is a bit unclear (we probably only need to ask), but a row in the excel sheet is a data related to a single file, and we don't know if multiple rows maps to the same cristin-id. So this code here is to catch that instance.

So what Joachim does here is correct.

**Review by anetteOlli — APPROVED** [review]

File: src/test/java/no/sikt/nva/UnisContentScrapingResultTest.java (hunk 3 above)
Comment (anetteOlli) [reply src/test/java/no/sikt/nva/UnisContentScrapingResultTest.java:104]: I can't see how this test does more than one thing? So I disagree with this comment.

**Pull request merged**